src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -93,9 +93,8 @@ public static class WebScriptHostBuilderExtension
 
                     if (!FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableDiagnosticEventLogging))
                     {
+                        // services that this depends on are registered at the webhost level
                         loggingBuilder.Services.AddSingleton<ILoggerProvider, DiagnosticEventLoggerProvider>();
```

Comment:
DiagnosticEventLoggerProvider is also registered in `WebHostServiceCollectionExtension.AddWebJobsScriptHost`. Do we need to register it here as well ?

Reply:
Yeah -- `ILoggerProviders` don't get shared between WebHost and JobHost. The way it works is:
- If the JobHost asks for service `IFoo` and there aren't any, it checks the parent WebHost for its version of `IFoo`. This is why a bunch of services can be shared between the two DI containers.
- However, logging requests an `IEnumerable<ILoggerProvider>` -- this exists in JobHost already, so it will not ask the WebHost for its version. `IOptions<T>` has the same limitation -- b/c .NET will always create you an `IOptions<T>`, it doesn't flow through to the parent at all. This has bitten us a bunch in the past. So there's certain types (loggers, options, etc) that don't flow from parent -> child like other types.